core/templates/pages/admin-page/jobs-tab/admin-jobs-tab.component.ts

```diff
@@ -0,0 +1,167 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the jobs tab in the admin panel.
+ */
+
+import { HttpClient } from '@angular/common/http';
+import { Component, EventEmitter, Output } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { AdminDataService } from '../services/admin-data.service';
+import { AdminPageConstants } from 'pages/admin-page/admin-page.constants';
+import { AdminPageData } from 'domain/admin/admin-backend-api.service';
+import { ComputationData } from 'domain/admin/computation-data.model';
+import { JobStatusSummary } from 'domain/admin/job-status-summary.model';
+import { Job } from 'domain/admin/job.model';
+
+interface AdminJobOutputResponse {
+  output: string[]
+}
+
+@Component({
+  selector: 'oppia-admin-jobs-tab',
+  templateUrl: './admin-jobs-tab.component.html'
+})
+export class AdminJobsTabComponent {
+  @Output() setStatusMessage: EventEmitter<string> = (
+    new EventEmitter
+  );
+  showingJobOutput: boolean;
+  jobOutput = [];
+  HUMAN_READABLE_CURRENT_TIME: string = '';
+  CONTINUOUS_COMPUTATIONS_DATA: ComputationData[] = [];
+  ONE_OFF_JOB_SPECS: JobStatusSummary[] = [];
+  UNFINISHED_JOB_DATA: Job[] = [];
+  AUDIT_JOB_SPECS: JobStatusSummary[] = [];
+  RECENT_JOB_DATA: Job[] = [];
+  constructor(
+    private httpClient: HttpClient,
+    private adminDataService: AdminDataService,
+    private urlInterpolationService: UrlInterpolationService,
+    private windowRef: WindowRef,
+  ) { }
+
+  showJobOutput(
+      jobId: string
+  ): void {
+    let adminJobOutputUrl = this.urlInterpolationService.interpolateUrl(
+      AdminPageConstants.ADMIN_JOB_OUTPUT_URL_TEMPLATE, {
+        jobId: jobId
+      });
+    this.httpClient.get(adminJobOutputUrl).toPromise()
+      .then((adminJobOutputResponse: AdminJobOutputResponse) => {
+        this.showingJobOutput = true;
+        this.jobOutput = adminJobOutputResponse.output || [];
+        this.jobOutput.sort();
+        document.querySelector('#job-output')
+          .scrollIntoView();
+      }, (errorResponse) => {
+        this.setStatusMessage.emit(
+          'Server error: ' + errorResponse.error
+        );
+      });
+  }
+
+  startNewJob(
+      jobType: string) : void {
+    this.setStatusMessage.emit('Starting new job...');
+    this.httpClient.post(AdminPageConstants.ADMIN_HANDLER_URL, {
```

Comment:
Can you move the http requests to a separate service.

Reply:
Done.

Reply:
Done.